crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -633,6 +625,16 @@ impl<'a, Provider: ResolverProvider> Resolver<'a, Provider> {
             }
 
             PubGrubPackage::Package(package_name, extra, None) => {
+                // If the dist is an editable, return the version from the editable metadata.
+                if let Some((_local, metadata)) = self.editables.get(package_name) {
```

Comment:
The only downside to all of this is that we now pay the cost of doing these lookups for all packages.